File: openmrs-client/src/main/java/org/openmrs/mobile/utilities/ThemeUtils.java
package org.openmrs.mobile.utilities;

import android.content.SharedPreferences;

import org.openmrs.mobile.application.OpenMRS;

public class ThemeUtils {

    public static boolean isDarkModeActivated() {
        SharedPreferences defaultSharedPref = OpenMRS.getInstance().getOpenMRSSharedPreferences();
        boolean isDarkMode = defaultSharedPref.getBoolean(ApplicationConstants.OpenMRSThemes.KEY_DARK_MODE, false);
        return isDarkMode;
    }

    public static void setDarkMode(boolean darkMode) {
        SharedPreferences.Editor editor = OpenMRS.getInstance().getOpenMRSSharedPreferences().edit();
        editor.putBoolean(ApplicationConstants.OpenMRSThemes.KEY_DARK_MODE, darkMode);
        editor.apply();
    }
}

File: openmrs-client/src/main/java/org/openmrs/mobile/utilities/ThemeUtils.kt
package org.openmrs.mobile.utilities

import com.openmrs.android_sdk.library.OpenmrsAndroid
import com.openmrs.android_sdk.utilities.ApplicationConstants

object ThemeUtils {

    @JvmStatic
    fun isDarkModeActivated(): Boolean {
        val defaultSharedPref = OpenmrsAndroid.getOpenMRSSharedPreferences()
        return defaultSharedPref.getBoolean(ApplicationConstants.OpenMRSThemes.KEY_DARK_MODE, false)
    }

    @JvmStatic
    fun setDarkMode(darkMode: Boolean) {
        val editor = OpenmrsAndroid.getOpenMRSSharedPreferences().edit()
        editor.putBoolean(ApplicationConstants.OpenMRSThemes.KEY_DARK_MODE, darkMode)
        editor.apply()
    }
}